src/common/telemetry/lib/Telemetry.cpp

```diff
@@ -0,0 +1,365 @@
+#include <cstring>
+#include <iostream>
+#include <thread>
+
+using namespace MAT;
+
+LOGMANAGER_INSTANCE
+
+namespace Telemetry
+{
+
+OsConfigLogHandle TelemetryManager::m_log = NULL;
+MAT::ILogger* TelemetryManager::m_logger = nullptr;
+bool TelemetryManager::m_initialized = false;
+
+TelemetryManager::~TelemetryManager() noexcept
+{
+    Shutdown();
+}
+
+void TelemetryManager::SetupConfiguration(bool enableDebug, int teardownTime)
+{
+    auto& logConfig = LogManager::GetLogConfiguration();
+
+    logConfig[CFG_BOOL_ENABLE_TRACE] = enableDebug;
+    logConfig[CFG_INT_TRACE_LEVEL_MIN] = 0;
+    logConfig[CFG_INT_MAX_TEARDOWN_TIME] = teardownTime;
+
+    m_logger = LogManager::Initialize(API_KEY);
+}
+
+bool TelemetryManager::Initialize(bool enableDebug, int teardownTime)
+{
+    if (m_initialized)
+    {
+        return true;
+    }
+
+    OsConfigLogInfo(m_log, "Initializing telemetry...");
+
+    try
+    {
+        if (NULL == m_log)
+        {
+            m_log = OpenLog("/var/log/osconfig_telemetry.log", NULL);
+        }
+
+        SetupConfiguration(enableDebug, teardownTime);
+
+        if (!m_logger)
+        {
+            return false;
+        }
+
+        LogManager::SetTransmitProfile(TransmitProfile::TransmitProfile_RealTime);
+        m_initialized = true;
+        OsConfigLogInfo(m_log, "Telemetry initialized successfully.");
+        return true;
+    }
+    catch (const std::exception& e)
+    {
+        if (m_log)
+        {
+            OsConfigLogError(m_log, "Failed to initialize telemetry: %s", e.what());
+        }
+        return false;
+    }
+}
+
+void TelemetryManager::EventWrite(Microsoft::Applications::Events::EventProperties event)
+{
+    if ((!m_initialized) || (!m_logger))
+    {
+        OsConfigLogError(m_log, "Telemetry not initialized");
+        return;
+    }
+
+    m_logger->LogEvent(event);
+
+}
+
+void TelemetryManager::Shutdown()
+{
+    if (!m_initialized)
+    {
+        OsConfigLogError(m_log, "Telemetry not initialized");
+        return;
+    }
+
+    OsConfigLogInfo(m_log, "Shutting down telemetry...");
+
+    try
+    {
+        LogManager::UploadNow();
+        std::this_thread::sleep_for(std::chrono::seconds(1)); // Without sleep, the upload may not complete
+        LogManager::FlushAndTeardown();
+        m_initialized = false;
+        OsConfigLogInfo(m_log, "Telemetry shutdown complete.");
+        CloseLog(&m_log);
+    }
+    catch (const std::exception& e)
+    {
+        if (m_log)
+        {
+            OsConfigLogError(m_log, "Exception during shutdown: %s", e.what());
+        }
+    }
+}
+
+bool TelemetryManager::ProcessJsonFile(const std::string& filePath)
+{
+    if (!m_initialized)
+    {
+        if (!Initialize())
+        {
+            OsConfigLogError(m_log, "Failed to initialize TelemetryManager");
+            return false;
+        }
+    }
+
+    OsConfigLogInfo(m_log, "Processing JSON file: %s", filePath.c_str());
+
+    FILE* file = fopen(filePath.c_str(), "r");
```

Comment:
use raii, either a class, smart pointer construct or an unwinder this may leak on throw.